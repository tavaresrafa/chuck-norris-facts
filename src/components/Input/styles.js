import styled from 'styled-components';
import { colors } from '../../styles/colors';

export const Container = styled.section`
  width: 100%;
  display: flex;
  justify-content: center;
  margin-bottom: 4rem;
`;

export const Input = styled.input`
  background: ${colors.white};
  border: 1px solid ${colors.gray2};
  font-size: 1.3rem;
  padding: 1.2rem;
  width: 100%;
  height: 4rem;
  text-align: center;

  @media (min-width: 576px) {
    width: 46rem;
    text-align: left;
  }
`;
