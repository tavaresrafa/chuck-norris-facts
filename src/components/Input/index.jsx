import React from 'react';
import * as S from './styles';

const Input = ({ placeholder, handleOnKeyDown, handleOnChange }) => (
  <S.Container>
    <S.Input
      defaultValue=""
      placeholder={placeholder}
      onKeyDown={handleOnKeyDown}
      onChange={handleOnChange}
    />
  </S.Container>
)

export default Input;
