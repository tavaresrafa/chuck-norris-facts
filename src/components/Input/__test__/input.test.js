import React from 'react';
import ReactDOM from 'react-dom';
import Input from '../../Input';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Input />, div);
});

// SNAPSHOTS
it('matches snapshot', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
});