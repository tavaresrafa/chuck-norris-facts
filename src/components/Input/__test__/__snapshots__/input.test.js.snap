// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="sc-AxjAm djeBuX"
>
  <input
    className="sc-AxirZ kPSsRL"
    defaultValue=""
  />
</section>
`;
